src/ocaml_helpers.h

```diff
@@ -27,7 +27,7 @@ static struct custom_operations binaryen_ops;
 #define Is_none(v) ((v) == Val_none)
 #define Is_some(v) Is_block(v)
 
-#define Safe_String_val(v) strdup(String_val(v))
+#define Safe_String_val(v) memcpy(malloc(caml_string_length(v) + 1), String_val(v), caml_string_length(v) + 1)
```

Comment:
There's a `+ 1` to include the string's null termination byte (since OCaml strings are null-terminated). Is it worth leaving a comment about that?

Also, I didn't think it was worth the hassle/trouble of having local variables in this macro to avoid calling `caml_string_length` twice, but let me know if you think otherwise.

Reply:
I think the comment would be helpful for anyone else looking at the code (so they don't need to track it back to this PR).

As for the variable: it's probably not needed.